### DISCUSSION TITLE
LUNA tokens stuck in Wormhole. What to do?

### DISCUSSION THREAD

**Discussion by cryptorus17** [wormhole-foundation/wormhole#277] [Problems / bugs] [answered]
I was using Wormhole to transfer 430.94 LUNA from Eth to Sollet wallet. The transaction was a "success" transferring my funds from Eth to this contract "0xf92cd566ea4864356c5491c177a430c222d7e678".
Although I'm not able to locate my LUNA in my Sollet wallet.

Eth Transaction hash 0xd099d465532b499987b8fe73922ec8dcb5d81c5997bfaa1489adfa82e539476b
Eth transfer from 0x0be833cF3c2B9d692c7859f06da9ef0e151B8a33
Transfer to (Sollet wallet) 4Uu6JW78DsfsJfV8N5vdRfPEtq41Vg972Y7E5fu2FKjK

Later that night I tried using Wormhole again, sending LUNA from Eth to Solana, same wallet, just a different amount. 
Received my LUNA to my Sollet wallet with no issues. 

https://etherscan.io/tx/0xd099d465532b499987b8fe73922ec8dcb5d81c5997bfaa1489adfa82e539476b (Not received) 
https://etherscan.io/tx/0x03fb27260c2def33bcbe21396cb00637cb4cb5b02fdefd21fb70a96b31489b53  (Received) 

<img width="1355" alt="Screen Shot 2021-07-28 at 7 36 50 AM" src="https://user-images.githubusercontent.com/88086309/127341982-3b83731a-f72e-42e8-a9ff-b61fc9751093.png">

I've been told by @ hendrikhofstadt (Github member) that I sent my LUNA to a UST wallet and that the Wormhole actually processed my transfer but given that the recipient address is not for the right token my LUNA tokens were not delivered and there is no way to recover them. 

How is this possible to transfer an asset using Wormhole to a wrong address??

On Wormhole, if I'm choosing to transfer an asset from Eth to Solana, I can't choose LUNA on Eth part and UST on Solana side, the moment i choose LUNA on the sending side the same asset automatically fills up on the receiving side. 
The Sollet wallet have the same receiving address for all tokens 4Uu6JW78DsfsJfV8N5vdRfPEtq41Vg972Y7E5fu2FKjK, be it ETH, UST, or LUNA. 

I see my 430.94 LUNA sit in the Wormhole contract as "Lock Assets" whatever that means, 
<img width="1147" alt="Screen Shot 2021-07-28 at 8 57 31 AM" src="https://user-images.githubusercontent.com/88086309/127356560-bf8f4a1d-ea5e-490d-b21e-f49e19c5020d.png">
https://etherscan.io/token/0xd2877702675e6ceb975b4a1dff9fb7baf4c91ea9?a=0xf92cd566ea4864356c5491c177a430c222d7e678

My Sollet wallet address: 
**4Uu6JW78DsfsJfV8N5vdRfPEtq41Vg972Y7E5fu2FKjK**

This is the only Sollet address that I have, and its impossible to send assets on Wormhole without connecting two wallets.
 

**So how is this possible that I received one transfer but not the other using the same method, same wallet, same address, same contract?**

or am I missing something here?

Thanks for your help

**Comment by albertjpn** [comment]


**Reply by hendrikhofstadt** [reply]
You used the Wormhole correctly, we're helping you complete the transfer (which got stuck due to a software bug we just fixed) as I'm typing this. The user above used the Wormhole incorrectly and sent funds to a wrong address.

**Reply by albertjpn** [reply]
Much appreciated! Very helpful!

**Comment by hendrikhofstadt** [comment, marked as answer]
@cryptorus17 I can only stress what I said in your issue. 

In your failed transaction, you specified an invalid recipient (different from the one that worked).

Working tx:
![image](https://user-images.githubusercontent.com/1405758/127495989-0ddcef90-e825-4d84-aa08-ec6200396ca8.png)

Failed tx:
![image](https://user-images.githubusercontent.com/1405758/127496107-c5466d89-40ef-4aa7-ae76-2a7ad6665440.png)

**Reply by cryptorus17** [reply]
> @cryptorus17 I can only stress what I said in your issue.
> 
> In your failed transaction, you specified an invalid recipient (different from the one that worked).
> 
> Working tx:
> ![image](https://user-images.githubusercontent.com/1405758/127495989-0ddcef90-e825-4d84-aa08-ec6200396ca8.png)
> 
> Failed tx:
> ![image](https://user-images.githubusercontent.com/1405758/127496107-c5466d89-40ef-4aa7-ae76-2a7ad6665440.png)

When you say invalid recipient, what does that mean? That I used a different Sollet address?

**Reply by cryptorus17** [reply]
Help me understand this, cause i have ZERO trust in the bridge right now. Im trying to understand how could this happen, so i wouldnt repeat it in the future.

**Reply by leoluk** [reply]
Please contact the developers of the wallet/UI you used. We did not build any of the UIs, all we know is what's on-chain.

**Reply by leoluk** [reply]
In case you used wormholebridge.com, the correct contact is Solana: https://github.com/solana-labs/oyster/issues

**Reply by cryptorus17** [reply]
But can you tell me what actually happened? What went wrong? Previously someone told me that i sent LUNA to UST address, which doesnt make much sense, because all assets on Solana use the same wallet address. 

There is not much UI, just connecting 2 wallets thats it.

**Reply by leoluk** [reply]
We don't know, please ask the developer of the UI you're using. There's tons of client side logic.

**Reply by leoluk** [reply]
(for example, the UI may have failed to create the token account, but still submit the tx - in which case they could add a retry feature that creates the account, but this is just speculation)
